File: lectures/week-3/exercisses/phonebook.c
#include <stdio.h>
#include <string.h>

typedef struct
{
    char name[20];
    char number[20];
} Person;

int main(void)
{
    Person people[4];
    char target[20];

    strcpy(people[0].name, "Alice");
    strcpy(people[0].number, "+1-123-456-7890");

    strcpy(people[1].name, "Bob");
    strcpy(people[1].number, "+1-987-654-3210");

    strcpy(people[2].name, "Charlie");
    strcpy(people[2].number, "+1-555-123-4567");

    strcpy(people[3].name, "David");
    strcpy(people[3].number, "+1-154-391-4275");

    printf("Name: ");
    scanf("%19s", target);

    for (int i = 0; i < 4; i++)
    {
        if (strcmp(people[i].name, target) == 0)
        {
            printf("Found:\n name: %s\n phone: %s\n", people[i].name, people[i].number);
            return 0;
        }
    }

    printf("Not found\n");
    return -1;
}

File: lectures/week-6/exercises/phonebook.py
people = [
    {"name": "Carter", "phone": "123-456-7890"},
    {"name": "David", "phone": "234-567-8901"},
    {"name": "John", "phone": "345-678-9012"}
]

name = input("Enter a name: ")

def find_person(name):
    for person in people:
        if person["name"] == name:
            return person
    return None

person = find_person(name)

if person:
    print(f"Phone: {person['phone']}")
else:
    print("Person not found.")